src/lib/RichText/Converter/ListLineBreaks.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\FieldTypeRichText\RichText\Converter;
+
+use EzSystems\EzPlatformRichText\eZ\RichText\Converter;
+use DOMDocument;
+use DOMXPath;
+
+/**
+ * Class ListLineBreaks.
+ *
+ * Processes lists below <literallayout> tag.
+ */
+class ListLineBreaks implements Converter
+{
+    /**
+     * For all <itemizedList> and <orderedList> below <literallayout>, move the list after the <literallayout>, so that it is not inside it.
+     *
+     * @param \DOMDocument $document
+     *
+     * @return \DOMDocument
+     */
+    public function convert(DOMDocument $document)
+    {
+        $xpath = new DOMXPath($document);
+        $xpathExpression = '//ns:literallayout [descendant::ns:orderedlist|descendant::ns:itemizedlist]';
+        $ns = $document->documentElement ? $document->documentElement->namespaceURI ?: '' : '';
+        $xpath->registerNamespace('ns', $ns);
+        $elements = $xpath->query($xpathExpression) ?: [];
+
+        // elements are list of <literallayout> elements
+        foreach ($elements as $element) {
+            $listItemToMove = [];
+            $listItemNo = 0;
+            /** @var \DOMElement $element */
+            foreach ($element->childNodes as $childNode) {
+                /** @var \DOMNode $childNode */
+                if ($childNode instanceof \DOMElement) {
+                    if ($childNode->tagName == 'orderedlist' || $childNode->tagName == 'itemizedlist') {
+                        $listItemToMove[$listItemNo] = $childNode;
+                        ++$listItemNo;
+                        continue;
+                    }
+                }
+            }
+
+            $paragraphNode = null;
+            foreach ($listItemToMove as $node) {
+                /** @var \DomNode $node */
+                if ($node->parentNode !== null) {
+                    $targetNode = $node->parentNode->parentNode;
+                    if ($targetNode !== null) {
+                        $targetNode->appendChild($node);
+                    }
+                }
+            }
+        }
+
+        return $document;
+    }
+}
```

Comment:
@vidarl I apologize for a bit chaotic review. I see now that it might be difficult to address it so I summarized my suggestions & more as the diff. Let me know if that works for you or if you have any questions.

```suggestion
use DOMDocument;
use DOMElement;
use DOMNode;
use DOMXPath;
use EzSystems\EzPlatformRichText\eZ\RichText\Converter;

/**
 * @internal
 *
 * Processes lists nested in <literallayout> tags.
 */
final class LiteralLayoutNestedList implements Converter
{
    private const FALLBACK_NAMESPACE = 'http://docbook.org/ns/docbook';
    private const ORDERED_LIST_TAG = 'orderedlist';
    private const ITEMIZED_LIST_TAG = 'itemizedlist';

    /**
     * For all <itemizedList> and <orderedList> nested in the <literallayout>, move the list after the <literallayout>,
     * so that it is not inside it.
     */
    public function convert(DOMDocument $document): DOMDocument
    {
        $xpath = new DOMXPath($document);
        $xpathExpression = '//ns:literallayout [descendant::ns:orderedlist|descendant::ns:itemizedlist]';
        $xpath->registerNamespace(
            'ns',
            null !== $document->documentElement && !empty($document->documentElement->namespaceURI)
                ? $document->documentElement->namespaceURI
                : self::FALLBACK_NAMESPACE
        );
        $elements = $xpath->query($xpathExpression) ?: [];

        // elements are list of <literallayout> elements
        /** @var \DOMElement $element */
        foreach ($elements as $element) {
            /** @var \DOMNode $childNode */
            foreach ($element->childNodes as $childNode) {
                if ($this->isNestedListNode($childNode)) {
                    $targetNode = $childNode->parentNode->parentNode;
                    if ($targetNode !== null) {
                        $targetNode->appendChild($childNode);
                    }
                }
            }
        }

        return $document;
    }

    /**
     * @phpstan-assert-if-true !null $childNode->parentNode
     */
    private function isNestedListNode(DOMNode $childNode): bool
    {
        return $childNode instanceof DOMElement
            && ($childNode->tagName === self::ORDERED_LIST_TAG || $childNode->tagName === self::ITEMIZED_LIST_TAG)
            && $childNode->parentNode !== null;
    }
}
```

Reply:
Fixed as suggested in 295b2127692c3b2b2aaa26c0c9310cb3c916a2f4